presto-main/src/test/java/io/prestosql/sql/query/TestDistinctAggregations.java

```diff
@@ -282,4 +282,14 @@ public void testUuidDistinct()
                 "             (UUID'be0b0518-35a1-4d10-b7f1-1b61355fa741')) AS t (uuid_col)",
                 "VALUES UUID'be0b0518-35a1-4d10-b7f1-1b61355fa741'");
     }
+
+    @Test
+    public void testIpAddressDistinct()
```

Comment:
was this failing before?

Reply:
It wasn't failing before because previously we were using `JoinCompiler` to generate code which is similar to `IpAddress#equals` operation.